### PR TITLE
Convert search store to an options store

### LINKED ISSUE #1258: Convert search store from a setup store to an options store
## Problem

A solution for the server state hydration problem is to convert the Pinia setup stores to options stores.

## Description

Search store needs to be converted into options store.

## Alternatives

#1254 describes an ad-hoc solution for server state hydration, however it means we would be using a non-standard implementation, and would probably face other problems in the future.

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1258 by @obulat

## Description

This PR converts the setup Pinia store into options store. This means that we pass an options object to the `defineStore` function that has `state`, `getters` and `actions`, similar to Vuex stores.

This PR also resets the state when homepage mounts so that the old search type is not used when navigating back to the homepage from a search page using the logo, and searching again.

## Testing Instructions

Try searching for all content, images and audios, opening single results, searching from the homepage

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
From a UI perspective, everything I tested is working correctly:

- Toggling and clearing filters
- SSR and client-side page loads
- Switching content types

File: src/pages/index.vue
Comment (obulat) [reply src/pages/index.vue:185]: Oh, wait. This will require the reset in the media store. Let me make an issue for fixing it after one of the store PRs is merged.

File: src/stores/search.ts
Comment (sarayourfriend) [inline src/stores/search.ts:49]: nit: This annotation isn't necessary, it's implicit in the type of `queryKeys`.

```suggestion
    (obj, key) => {
```

File: src/stores/search.ts
Comment (sarayourfriend) [inline src/stores/search.ts:101]: Should we choose to use either `this` or the explicit `state` parameter consistently? Mixing the two is confusing.

File: test/unit/specs/stores/search-store.spec.js
Comment (sarayourfriend) [inline test/unit/specs/stores/search-store.spec.js:385]: Why this change?

File: test/unit/specs/stores/search-store.spec.js
Comment (sarayourfriend) [inline test/unit/specs/stores/search-store.spec.js:415]: Why remove this test?

**Review by sarayourfriend — COMMENTED** [review]
I found a few bugs testing this.

https://github.com/WordPress/openverse-frontend/issues/1262
https://github.com/WordPress/openverse-frontend/issues/1263
https://github.com/WordPress/openverse-frontend/issues/1264
https://github.com/WordPress/openverse-frontend/issues/1265

None of them are caused by this branch though :tada: 

Just a couple questions about the tests but should be able to approve after that :grin:

File: test/unit/specs/stores/search-store.spec.js
Comment (obulat) [reply test/unit/specs/stores/search-store.spec.js:385]: Normally, `All Content` view only has 12 filters available (License, License Types, Search By Creator), and the tests should show 12 filters. But in tests, for some reasons, the watcher was probably not firing, and thus it was showing 37 filters on 'All Content' view.
I've added the tests starting on lines 412 to make sure that we do fix this problem in the future.
After the conversion of the search store to options store, the filters are changed correctly, so I removed the now redundant tests. The `image` -> `all_media` change that is tested on line 385 shows that the correct number of filters is shown.

Note: I did not see this incorrect number of filters in the live app, only in tests.

Comment (sarayourfriend) [reply test/unit/specs/stores/search-store.spec.js:385]: I see! Thanks for explaining.

File: test/unit/specs/stores/search-store.spec.js
Comment (sarayourfriend) [reply test/unit/specs/stores/search-store.spec.js:415]: Explained here: https://github.com/WordPress/openverse-frontend/pull/1259#discussion_r849371816

File: src/stores/search.ts
Comment (sarayourfriend) [reply src/stores/search.ts:101]: Ah interesting. I guess it's fine to mix them, I will get used to it!

**Review by krysal — APPROVED** [review]
Everything seems to work as expected 👍

**Pull request merged**
